runtime-sdk/src/modules/core/mod.rs

```diff
@@ -484,11 +493,22 @@ impl<Cfg: Config> Module<Cfg> {
     /// in the context's method registry. Transactions that fail still use gas, and this query will
     /// estimate that and return successfully, so do not use this query to see if a transaction will
     /// succeed.
-    #[handler(query = "core.EstimateGas")]
+    #[handler(query = "core.EstimateGas", allow_private_km)]
     pub fn query_estimate_gas<C: Context>(
         ctx: &mut C,
         mut args: types::EstimateGasQuery,
     ) -> Result<u64, Error> {
+        let mut extra_gas = Self::get_estimate_gas_extra(ctx);
+        // In case the runtime is confidential we are unable to authenticate the caller so we must
+        // make sure to zeroize it to avoid leaking private information.
+        if ctx.is_confidential() {
+            args.caller = Some(
+                args.caller
+                    .unwrap_or(CallerAddress::Address(Default::default()))
```

Comment:
:information_source: for sapphire we'll have to set `caller: Some(CallerAddress::EthAddr(some_dummy_value))` from the client

Reply:
This is exactly what the `.zeroized()` call does ;-)

Reply:
Fixed, please take a look.

Reply:
lg